### PR TITLE
Add table `aws_ssmincidents_response_plan`. Closes #1821

### LINKED ISSUE #1821: Add table aws_ssmincidents_response_plan
**References**

https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.297/service/ssmincidents#GetResponsePlanOutput

Note: This table is required for new query develoment.

### PR DESCRIPTION
**Pull request by madhushreeray30**
# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```sql
select name, arn, region, account_id from aws_ssmincidents_response_plan
+--------------------+----------------------------------------------------------------------+----------------+--------------+
| name               | arn                                                                  | region         | account_id   |
+--------------------+----------------------------------------------------------------------+----------------+--------------+
| response-plan-test | arn:aws:ssm-incidents::111111111111:response-plan/response-plan-test | ap-southeast-1 | 111111111111 |
| response-plan-test | arn:aws:ssm-incidents::111111111111:response-plan/response-plan-test | us-east-1      | 111111111111 |
+--------------------+----------------------------------------------------------------------+----------------+--------------+
```

```sql
select name, display_name, arn, incident_template from aws_ssmincidents_response_plan where arn = 'arn:aws:ssm-incidents::111111111111:response-plan/response-plan-test' and region = 'ap-southeast-1'
+--------------------+--------------------+----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
| name               | display_name       | arn                                                                  | incident_template                                                                                                                  |
+--------------------+--------------------+----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
| response-plan-test | response-plan-test | arn:aws:ssm-incidents::111111111111:response-plan/response-plan-test | {"DedupeString":null,"Impact":5,"IncidentTags":null,"NotificationTargets":null,"Summary":"summaryyy","Title":"response-plan-test"} |
+--------------------+--------------------+----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+

```
</details>

### CODE REVIEW

**Comment by misraved** [comment]
Naming convention reference - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssmincidents_response_plan

File: docs/tables/aws_ssmincidents_response_plan.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  chat_channel,
+  incident_template,
+  integrations,
+  title
+from
+  aws_ssmincidents_response_plan
+where
+  chat_channel is not null;
+```
+
```
Comment (ParthaI) [inline docs/tables/aws_ssmincidents_response_plan.md:39]:
```suggestion
```

**Review by ParthaI — COMMENTED** [review]
@madhushreeray30, Please take a look at the review comments, thanks!

**Review by ParthaI — APPROVED** [review]
LGTM

**Pull request merged**